app/helpers/state_bar_helper.rb

```diff
@@ -0,0 +1,91 @@
+module StateBarHelper
+  Colors = {
+    intermediate: %i[draft],
+    rejected: %i[aborted lost refused scrapped sold],
+    validated: %i[done finished given in_use invoice ongoing order repaid won]
+  }.flat_map { |key, values| values.map { |v| [v, key] } }
```

Comment:
Could this be moved to a method of the `Hash` class? (`reverse`? `flip`? Maybe `transpose`, similarly to `Array#transpose`)

Reply:
I don't think moving this to the Hash class is a good idea as the implementation is really specific: what about if values are not unique?
What we will get if we add sth like `Hash#transpose` is a `MultiMap` and not a `Hash`

Reply:
The target of this would be a category of `Hash`. The implementation would raise if there were non-unique values.
Ex in the Ruby Standard Lib: `Array#transpose` raises if the element array are not all of the same length.
```ruby
irb(main):002:0* [[:a, :b, :c], [:a, :b]].transpose
IndexError: element size differs (2 should be 3)
        from (irb):2:in `transpose'
```

Reply:
heh, ok